src/rai_hmi/rai_hmi/text_hmi.py

```diff
@@ -30,103 +32,107 @@
 )
 from langchain_openai.chat_models import ChatOpenAI
 from PIL import Image
+from pydantic import BaseModel
+from streamlit.delta_generator import DeltaGenerator
 
 from rai.agents.conversational_agent import create_conversational_agent
 from rai.agents.state_based import get_stored_artifacts
 from rai.messages import HumanMultimodalMessage
-from rai.node import RaiBaseNode
-from rai.tools.ros.native import GetCameraImage, Ros2GetTopicsNamesAndTypesTool
 from rai_hmi.base import BaseHMINode
+from rai_hmi.task import Task
 
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
 logger = logging.getLogger(__name__)
 
-
 st.set_page_config(page_title="LangChain Chat App", page_icon="🦜")
-robot_description_package = sys.argv[1] if len(sys.argv) > 1 else None
-
-if robot_description_package:
-    st.title(f"{robot_description_package.replace('_whoami', '')} chat app")
-else:
-    st.title("ROS 2 Chat App")
-    logging.warning(
-        "No robot_description_package provided. Some functionalities may not work."
-    )
-
-
-@st.cache_resource
-def initialize_ros_node(robot_description_package: Optional[str]):
-    rclpy.init()
-
-    node = BaseHMINode(
-        f"{robot_description_package}_hmi_node",
-        robot_description_package=robot_description_package,
-    )
 
-    return node
+MODEL = "gpt-4o-mini"
 
 
-def decode_base64_into_image(base64_image: str):
-    image = Image.open(io.BytesIO(base64.b64decode(base64_image)))
-    return image
+# ---------- Cached Resources ----------
+@st.cache_resource
+def parse_args():
+    robot_description_package = sys.argv[1] if len(sys.argv) > 1 else None
+    return robot_description_package
 
 
 @st.cache_resource
 def initialize_agent(_node: BaseHMINode):
     llm = ChatOpenAI(
         temperature=0.5,
-        model="gpt-4o-mini",
+        model=MODEL,
         streaming=True,
     )
-    rai_node = RaiBaseNode(node_name="__rai_node__")  # this is so wrong
-    tools = [Ros2GetTopicsNamesAndTypesTool(node=_node), GetCameraImage(node=rai_node)]
```

Comment:
Can we please leave it?

Reply:
https://github.com/RobotecAI/rai/pull/188/commits/798f84e0a122f02564840aec813732e67e56174d

Reverted